packages/nc-gui-v2/composables/useTheme/index.ts

```diff
@@ -12,36 +12,81 @@ const [setup, use] = useInjectionState((config?: Partial<ThemeConfig>) => {
   const primaryColor = useCssVar('--color-primary', typeof document !== 'undefined' ? document.documentElement : null)
   const accentColor = useCssVar('--color-accent', typeof document !== 'undefined' ? document.documentElement : null)
 
-  /** current theme config */
-  const currentTheme = useStorage<ThemeConfig>(
-    `${NOCO}db-theme`,
-    {
+  const { project, getProjectMeta, updateProject } = useProject()
+
+  const route = useRoute()
+
+  // set theme based on active project
+  watch(project, () => {
+    setTheme({
       primaryColor: themeV2Colors['royal-blue'].DEFAULT,
       accentColor: themeV2Colors.pink['500'],
+      ...getProjectMeta()?.theme,
+    })
+  })
+
+  // set default theme for routes out of project
+  watch(
+    () => route,
+    (v) => {
+      if (!v.params?.projectId) {
+        setTheme({
+          primaryColor: themeV2Colors['royal-blue'].DEFAULT,
+          accentColor: themeV2Colors.pink['500'],
+        })
+      }
     },
-    localStorage,
-    { mergeDefaults: true },
+    { deep: true },
```

Comment:
don't use `deep` - it's usually an indicator that you don't know what you're actually watching.
In this case just watch `() => route.params`